src/components/search/SearchResults.tsx

```diff
@@ -0,0 +1,88 @@
+import Fuse from "fuse.js";
+import { useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+import PostCard from "../PostCard";
+import LoadingSpinner from "../ui/LoadingSpinner";
+
+export type ResultObj = {
+  slug: string;
+  data: {
+    title: string;
+    description: string;
+    date: string;
+  };
+};
+
+type SearchResult = {
+  item: ResultObj;
+  refIndex: number;
+  score: number;
+};
+
+const DEFAULT_NUM = 5;
+
+const SearchResults = ({ query }: { query: string }) => {
+  const [matchingResults, setMatchingResults] = useState<SearchResult[]>([]);
+  const [numOfResults, setNumOfResults] = useState(DEFAULT_NUM);
+  const { status, error, data } = useQuery<SearchResult[]>({
+    queryKey: ["search"],
+    queryFn: async () => {
+      try {
+        const res = await fetch(`/postSearch.json`);
+        if (!res.ok) throw new Error("Network response was not ok");
+        return res.json();
+      } catch (e) {
+        console.error(e);
+        return e;
+      }
+    },
+  });
+
+  const fuse = new Fuse(data, {
```

Comment:
Super cool! I've never used this before but looks like a nice API 🔥

Reply:
so mine is the guinea pig? 😂

Reply:
lol, well if it means anything, I implemented this on another blog this week first. So you’re the second guinea pig? lol And then I'll do mine tomorrow probably for my blog. Just bundling over here

Reply:
LOL mostly kidding but nice! Feeling good then!